### ISSUE TITLE
Projection handling poorly documented for ol.source.Vector

### ISSUE THREAD

**Issue by Barryrowe** [openlayers/ol3#4264]
I'm working through some simple examples, and I've found that for some reason valid GeoJSON data that can be read as features inline, or manually loaded into a layer via an AJAX call, fails to load and render properly when using 

    new ol.source.Vector({
        url: 'my-url/here.geojson',
        format: new ol.format.GeoJSON()
    })

It works fine when I use the example geoJson from the vector layer example: http://openlayers.org/en/master/examples/vector-layer.html

However, when I load in a custom geoJSON file through the url option, it never renders the features. I can debug and see the load featureloader.xhr succeeds and gets the text properly. I can even see it properly parse the features and validate they get added to the source. I also watched the projection at that point, and it is reflecting that set on the View. However, the features don't ever render.

I've created a repro here in this plnkr: http://plnkr.co/edit/Sw7kIad6NnxrUf6dfDdD

It creates 3 layers with the data in question:

1. does so inline, where the data is an inline literal javascript object
2. attempts to use the url option on a Vector source (should fill in yellow circles for the 2 points but doesn't)
3. manually makes an ajax call and adds calls readFeatures from the result (does so if you click the button)

I'm hoping this is something with my data, but it appears that the url option on ol.source.Vector is just partially broken

**Comment by ahocevar** [comment]
It's your data, because the GeoJSON does not advertise the CRS of the coordinates it contains. But you can work around that by overriding the default data projection (which is EPSG:4326) of the format:
```
format: new ol.format.GeoJSON({
  defaultDataProjection: 'EPSG:3857'
}),
```

**Comment by Barryrowe** [comment]
@ahocevar This does not solve the problem. I updated the plnkr to provide a defaultDataProjection on the source of the layer loading via url, and it still does not render in the features.

You can see in the plnkr that the inline version of the Layer also does not have a CRS property on the FeatureCollection and does not provide a defaultDataProjection on it's format, but loads without issue.

I can also manually load the data via ajax and create a layer like below and that works without setting defaultDataProjection.

     $.getJSON('two-points.geojson', function (data) {
              var remoteLayer = new ol.layer.Vector({
                  source: new ol.source.Vector({
                      features: (new ol.format.GeoJSON()).readFeatures(data)
                  }),
                  style: manualAjaxStyleFunction
              });
              map.addLayer(remoteLayer);
          });

You can see that setting all scenarios in the plunkr: http://plnkr.co/edit/Sw7kIad6NnxrUf6dfDdD
you can click the button to manually load the data via ajax.

the resulting styles should show:
--Inline layer:  draws points as red circles (no fill)
--URL property layer: should draw points as yellow filled circles (these do not render)
--Manually Loaded via AJAX: smaller blue filled circles

**Comment by ahocevar** [comment]
`defaultDataProjection` is an option on the format, not the source. Your snippet above works because it does not transform from data to view projection when reading.

**Comment by ahocevar** [comment]
I updated your Plunker accordingly: http://plnkr.co/edit/Y6AKVVhHVmLlrAbe01SZ

**Comment by Barryrowe** [comment]
@ahocevar I see my mistake (I had set the format right, but accidentally commented out the addition of the remoteloading layer...super silly mistake).

Can you provide some insight as to why you have to provide the projection option for the format when the data is loaded via url property, but when the data is loaded inline it isn't required?

It's the exact same data, but this works fine without the projection option on the format:

     //Inline version of data, and layer
        //----THIS ONE WORKS
        var geojsonObject = { "type": "FeatureCollection",
            "features": [
              { "type": "Feature",
                  "geometry": {"type": "Point", "coordinates": [-9530865.926035, 4612459.099040]},
                  "properties": {"id": "ABCDEF", "tlm_number": "-------", "location_description":"BOGUS", "phase": "A", "normal_state": "Something", "substation_name": "BOGUS", "circuit": "#111", "underground_circuit_number": "BLAH", "construction_type": "None", "facility_status": "None", "company_number": "CompanyNumber", "FUSE_SIZE_1": "65"}
              },
              { "type": "Feature",
                  "geometry": {"type": "Point", "coordinates": [-9531160.278699, 4612415.157190]},
                  "properties": {"id": "ABCDEFG", "tlm_number": "-------", "location_description": " Bogus ", "phase": "A", "normal_state": "Something", "substation_name": "BOGUS", "circuit": "#112", "underground_circuit_number": "None", "construction_type": "None", "facility_status": "One", "company_number": "CompanyNumber", "FUSE_SIZE_1": "65"}
              }
            ]
        };
        var inlineVectorLayor = new ol.layer.Vector({
            source: new ol.source.Vector({
                features: (new ol.format.GeoJSON()).readFeatures(geojsonObject)
            }),
            style: styleFunction
        });

**Comment by ahocevar** [comment]
When you configure a source with features, you have to provide them in the view projection. Since your GeoJSON already has the features in the view projection, it will work fine.

When you configure a source with a url and a format, the feature loader gets the CRS from the data and transforms the features to the view projection. In your case, the GeoJSON does not advertise the CRS of the coordinates, so the default (EPSG:4326) is used. So the feature loader transforms the data from EPSG:4326 to the view projection (EPSG:3857), which results in incorrect coordinates, because your data is in a different projection that is not advertised.

Make sense?

**Comment by Barryrowe** [comment]
@ahocevar Yes and No. What you are saying makes sense, it just doesn't seem consistent. 

I am not sure I understand how the inline version works if we're not telling it what projection the features are in when we add them to the source. It's the exact same JSON as being loaded by the remote. Shouldn't it be trying to convert the feature data from the default (EPSG:4326) to the View's projection as well since that layer's source's format doesn't define a projection and the data still doesn't have the CRS?

**Comment by Barryrowe** [comment]
@ahocevar I guess what you're saying is if a source has its ```features``` property set directly, the assumption is that the features are already "translated" to the View projection, and no translation is attempted, but if the ```url``` property is set, this assumption is thrown out and it expects to interpret the featureCollection's projection from the data, or assumes the default of (EPSG:4326) and always attempts a translations to the view projection.

If this is the case, I feel like one or the other should be the case in ALL instances of creating a new Source object.

**Comment by ahocevar** [comment]
The source caches features in the view projection. This is for performance reasons, so they do not have to be transformed whenever they are rendered.

When you pass features to the source, it does not know about the format, so it does not know about the data projection. When you configure a source with a url, it has the chance to transform the features while parsing them, because the renderer requests features from the source. Maybe this should be better documented. Your above comment summarises it quite well. Do you want to create a pull request with a documentation improvement?

**Comment by Barryrowe** [comment]
Thanks for all the info. I agree the documentation should mention the difference. It was very much unexpected from a consumer point of view. 

I will try and get a PR submitted for that soon. I haven't dug through the source to see how the API docs are built. Are they driven by the jsdoc files, or is everything under the doc directory?

**Comment by ahocevar** [comment]
The API docs are generated by jsdoc3. The config options you'll want to improve documentation for are in [`externs/olx.js`](https://github.com/openlayers/ol3/blob/v3.10.1/externs/olx.js#L4993-L5104).

**Comment by Barryrowe** [comment]
Just so I make sure I have this right, when you do:

    (new ol.format.GeoJSON()).readFeatures(someDataIn900913Format)

Since the format has no defined default projection, and we don't pass anything to the 2nd param of readFeatures, it just brings over the coordinates of the features EXACTLY as represented in the data passed in, or in other words "the developer assumes the data is already in the right projection coordinate system".

And in ```featureloader.js``` when it runs:
     var features = format.readFeatures(source,
                      {featureProjection: projection});
                  success.call(this, features);

The second parameter is telling ```readFeatures()``` to attempt a translation from the format's projection, to the presented projection, which will translate the coordinates on the passed in feature data.

**Comment by ahocevar** [comment]
Yes. In your first snippet above, you could also trigger coordinate transformation by passing a featureProjection:
```js
(new ol.format.GeoJSON()).readFeatures(someDataIn900913Format, {featureProjection: 'EPSG:3857'})
```

**Comment by Barryrowe** [comment]
Awesome, thanks again for all the info, and taking the time with someone who's brand new to the whole GIS world. This is making a lot more sense now.

**Comment by ahocevar** [comment]
Fixed with #4273. Thanks for the improvement @Barryrowe.
